doc/rst/language/spec/tuples.rst

```diff
@@ -714,6 +715,286 @@ where a comma-separated list of components is valid.
+
+Consider the following example:
+
+   .. code-block:: chapel
+
+      record R { var x: int; }
+
+      var a: [0..0] int;
+      var i: int;
+      var r: R;
+
+      // The int `i` is copied when captured, but `a` and `r` are not.
+      test((a, i, r));
+
+      // Modify the globals, then print the tuple.
+      proc test(tup) {
+        a[0] = 1;
+        i = 2;
+        r.x = 3;
+
+        // Outputs (1, 0, (x = 3)).
+        writeln(tup);
+      }
+
+   .. BLOCK-test-chapeloutput
+
+      (1, 0, (x = 3))
+
+The tuple expression ``(a, i, r)`` will capture the array ``a`` and the record
+``r`` by `ref`, but will create a copy of the integer ``i``.
+
+   *Rationale*
+
+   Tuple expressions and other forms of light tuple are designed to act like
+   a lightweight bundle of arguments. They behave similarly to the
+   individual arguments of a routine call.
+
+   It would be prohibitively expensive for some argument types (such as
+   arrays) to be copied by default when passed as an argument to a
+   routine call.
+
+   The same logic applies to tuple expressions. When the default argument
+   intent of a value's type is some form of `ref`, a tuple expression will
+   capture the value by reference in order to avoid a potentially
+   expensive copy operation.
+
+Tuple Variable Behavior
+-----------------------
+
+Tuple variables are a form of heavy tuple. Like other heavy tuples, tuple
+variables will copy elements as though the element is an argument with
+the `in` intent.
+
+   *Example (tuple-variable-behavior.chpl)*.
+
+For example, in this code:
+
+   .. code-block:: chapel
+
+      record R { var x: int; }
+
+      var a: [0..0] int;
+      var i: int;
+      var r = new R(0);
+
+      // The tuple variable `tup` copies `a`, `i`, and `r`.
+      var tup = (a, i, r);
+
+      a[0] = 1;
+      i = 2;
+      r.x = 3;
+
+      // This will output (0, 0, (x = 0)).
+      writeln(tup);
+
+   .. BLOCK-test-chapeloutput
+
+      (0, 0, (x = 0))
+
+The tuple variable ``tup`` will make a copy of the array ``a``, the record
+``r``, and the module integer ``i``.
+
+When these three variables are subsequently modified, changes to them are
+not reflected in ``tup`` when it is written to standard output.
+
+.. _Tuple_Argument_Behavior:
+
+Tuple Argument Behavior
+-----------------------
+
+A tuple argument to a routine may be either a light tuple or a heavy tuple
+depending on its argument intent.
+
+If the tuple argument has the default argument intent, then it is a light
+tuple and some of its elements may be captured by `ref` depending on
+their default argument intent.
+
+If the tuple argument has the `in` intent, then it is a heavy tuple and all
+of its elements are captured by value, as though each element has the `in`
+intent.
+
+If a routine argument is a tuple with the default argument intent and a
+heavy tuple (such as a tuple variable) is passed to it, the heavy tuple
+will be silently converted into a light tuple.
```

Comment:
"silently converted" -> "implicitly converted"

Add "The resulting light tuple may refer to elements in the original heavy tuple"